### PR TITLE
Fixes a couple of small issues

### PR DESCRIPTION
**Pull request by adamretter**
1. Incorrectly colouring pack sizes in the GUI when first calculating composite packs
2. Incorrect URL to file path decoding. Previously did not support file names that had a `+` character in them.

As previously discussed here: https://izpack.atlassian.net/browse/IZPACK-1582

### CODE REVIEW

**Comment by adamretter** [comment]
I don't really understand why Travis-CI is failing as running the same Maven targets on my machine here everything passes. Perhaps one of the IzPack devs could comment?

**Comment by rkrell** [comment]
You're welcome to contribute.

I made some comment in the source code changes.

One general note:
Please don't mix several issues into one big pull request. It is very hard to review, discuss and understand your changes and their background, especially, like for this, if there is not explained for exampe in JIRA, what happened to you to make it necessary to fix.

Regarding the byte checks in the Packer and Unpacker - this is a quite critical change for the case of whatever mistake and I'm afraid of this, to be honest. Did you test this for all types of compression IzPack supports currently? Thus building installers with compressed packs and launch them?

Furthermore I don't understand how this change depends on the discussion in [IZPACK-1582](https://izpack.atlassian.net/browse/IZPACK-1582). The only point regarding this issue which you intend to fix herewith is commit e4001f3 (which is what the test makes failing).

File: izpack-core/src/test/java/com/izforge/izpack/merge/resolve/MergeableResolverTest.java [hunk 1]
```diff
@@ -77,6 +77,15 @@ public void testGetMergeableWithSpaces() throws Exception
         assertThat(mergeable, MergeMatcher.isMergeableContainingFile("com/sora/panel/VimPanel.class"));
     }
 
+    @Test
+    public void testGetMergeableWithPlusSign() throws Exception
+    {
+        resource = ClassLoader.getSystemResource("com/izforge/izpack/merge/test/vim-panel-1.0-SNAPSHOT+201804141646.jar");
+        resource = new File(FileUtil.convertUrlToFilePath(resource) + "!com").toURI().toURL();
```
[the remark this answers is not in the document]
Comment (adamretter) [reply izpack-core/src/test/java/com/izforge/izpack/merge/resolve/MergeableResolverTest.java]: Ah right I missed a file from the commit. I have now added that back in, thanks.

**Comment by adamretter** [comment]
@rkrell This PR contains very small changes in total. They are 3 fold, the first 2 of which are in the IZPACK-1582 ticket:

1. Fixing the `+` chars in the filename.
2. Investigating why compression doesn't work (although not yet fixed).
3. Fixing a small UI bug that I found along the way (just a one liner).

Regards the checks in Packer and Unpacker. I added them because compression of any sort is not working for me in IzPack 5 as I reported in the IZPACK-1582. I think these checks show the problem, at least I now get more sensible exceptions and error messages when installing from compressed packs. Maybe I should move those checks into a separate PR, and we can try and fix compression in there. Are there no unit tests for compression/decompression of packs?

**Comment by rkrell** [comment]
I miss an important thing:

What is wrong for you with compression? Which version (release, GIT HEAD, snapshot from snapshot repository)? Is there a stacktrace, log entry, whatever effect? It's better to share these details.
I ask because after the new implementation I saw running compiling and running installers for compressed packs for me. Maybe something has broken on the way to the latest snapshot, maybe just a migration problem? Can you track this problem back for 5.1.2, 5.1.1, 5.1.0?
I'd appreciate a valid example showing the misfunction first. This might take slightly more time than writing the few lines of code, but it is transparent for the guys who decide about code changes and releases and helps them avoiding wrapping up every single issue from ground up.
Such things are better discussed in a thread belonging to one and only one issue.
That's what I wanted to tell.

And no: There doesn't exist any good integration test, not so good.
There should be also a better example in the Wiki. Much work, not all done.

**Comment by rkrell** [comment]
> I don't understand how I could have been any clearer, I started a discussion on this in July 2017 in the Jira - https://izpack.atlassian.net/browse/IZPACK-1582. I even posted version numbers and stack traces showing that the compression doesn't work. 
Ok, I missed that in the issue.

> The compression issue was completely ignored until now.
Sorry to say, but You should be a bit more friendly to the team, as we try to be.
The issue wasn't ignored, but nobody has time to solve it. The rate of contributions is quite small and we are all busy with our jobs we get paid for.
We are just a small number of volounteers and no personal support. The usual debates in the open source world...

**Comment by adamretter** [comment]
@rkrell Sorry I wasn't trying to be unfriendly. I also work on a large open source project. I completely understand the not-enough time thing and not paid etc. However, I was always willing to contribute but I could not seem to get a decent dialog started. What I meant is - I am thankful that people responded about the filename issue, however there really was no response about the compression issues.

I look forward to working with you on this and contributing fixes for this...

**Comment by rkrell** [comment]
@adamretter Your issue is not the only issue nobody responded to. If nobody is able to deal with is, what should be written there? "Thanks for using IzPack. We are working night and day to get your problem solved."? :-/ Not getting a response does not mean ignoring it, new issues don't get lost.
And regarding the not started dialog - I disagree. The dialog was started. The only wish has been to follow some rules, to make it possible to
- review your changes
- making a changelog for the next release
I find it quite usual for new contributors to send one PR for one single problem, should not be that hard.
Mostly, for not getting lost in code diffs, we also create JIRA issues, to refer to them in the release announcement. This is the style IzPack is based on. Every open source problem has more or less rules how are issues reported, how to code or to document and so on.
Even if the code changes are small they can break something, and as a reviewer I won't dig into single files and lines of code of a PR breaking something, but revert the PR as a whole while returning it to the contributor to fix it. In your case there would get lost changes which might be fine. I hope you understand this.

**Comment by adamretter** [comment]
@rkrell I am not sure I follow you exactly.

What do I need to do to help you progress both of my PR's? Happy to discuss separately in each PR respectively.

**Comment by rkrell** [comment]
As already said, I'd appreciate a small, compilable example showing the compression misfunction, which we will be able to discuss, instead of a big installer doing dozens of other things. Are you able to deliver it?

**Comment by dizzzz** [comment]
@adamretter thnx for your investigation and the PR. I hope it can be pulled in and released soonish as it will allow the project to continue....

**Comment by joewiz** [comment]
Yes, would it be possible for this to be reviewed? It's an important fix for us.

**Comment by duncdrum** [comment]
any progress on this ?

**Comment by rkrell** [comment]
Thanks for the fixes.

**Pull request merged**